### PR TITLE
Rent model

### PR DESCRIPTION
**Pull request by fcavazzoli**
## Rent Model
this ranch implements the rent model given one user, one book a start renting date and the end date.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,18 +10,18 @@
...
   enable_extension "plpgsql"
 
   create_table "books", force: :cascade do |t|
-    t.string "genre"
-    t.string "author"
-    t.string "image"
-    t.string "title"
-    t.string "editorial"
-    t.string "year"
+    t.string "genre", default: "Folklore", null: false
```
Comment (floresv) [inline db/schema.rb]: default?

File: app/models/rent.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Rent < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id', inverse_of: false
```
Comment (floresv) [inline app/models/rent.rb]: Do you really need `foreign_key: 'user_id', inverse_of: false` ? Think about it, please.

File: db/migrate/20190507152056_create_rents.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateRents < ActiveRecord::Migration[5.1]
+  def change
+    create_table :rents do |t|
+      t.belongs_to :user
+      t.belongs_to :book
+      t.date :startDate
```
Comment (floresv) [inline db/migrate/20190507152056_create_rents.rb]: Rails prefer [snake_case](https://guides.rubyonrails.org/active_record_basics.html#schema-conventions)

File: log/development.log [hunk 4]
```diff
@@ -5,3 +5,490 @@
   [1m[35m (0.1ms)[0m  [1m[35mCOMMIT[0m
   [1m[35m (0.1ms)[0m  [1m[34mSELECT pg_advisory_unlock(1580829076522303230)[0m
   [1m[35m (0.1ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+  [1m[35m (0.1ms)[0m  [1m[34mSELECT pg_try_advisory_lock(1580829076522303230)[0m
```
Comment (floresv) [inline log/development.log]: Why?

File: spec/factories/rents.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :rent do
+    user { '' }
+    book { '' }
```
Comment (floresv) [inline spec/factories/rents.rb]: mmm... please add factory [associations](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations).

File: spec/models/rent_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (floresv) [inline spec/models/rent_spec.rb]: Do you need this file? remember, We are using rspec.

File: spec/models/rent_spec.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+require 'rails_helper'
+
+RSpec.describe Rent, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (floresv) [inline spec/models/rent_spec.rb]: WIP?

File: log/development.log (hunk 4 above)
Comment (fcavazzoli) [reply log/development.log]: it is in the gitignore, i dont know why it is still pushing !

File: spec/models/rent_spec.rb (hunk 7 above)
Comment (fcavazzoli) [reply spec/models/rent_spec.rb]: yes :)

File: spec/factories/rents.rb (hunk 5 above)
Comment (fcavazzoli) [reply spec/factories/rents.rb]: did it

File: db/schema.rb (hunk 1 above)
Comment (fcavazzoli) [reply db/schema.rb]: solved

File: app/models/rent.rb (hunk 2 above)
Comment (fcavazzoli) [reply app/models/rent.rb]: removed foregin_key, rubocop demands inverse_of

File: app/models/rent.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Rent < ApplicationRecord
+  belongs_to :user, inverse_of: false
+  belongs_to :book, inverse_of: false
```
Comment (floresv) [inline app/models/rent.rb:5]: Why `inverse_of`? 🤔

File: db/migrate/20190507152056_create_rents.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+class CreateRents < ActiveRecord::Migration[5.1]
+  def change
+    create_table :rents do |t|
+      t.belongs_to :user
+      t.belongs_to :book
```
Comment (floresv) [inline db/migrate/20190507152056_create_rents.rb]: If it's null? 🤔

File: spec/factories/rents.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :rent do
+    association :user, factory: :user
+    association :book, factory: :book
+    start_at { '2019-05-07' }
+    end_at { '2019-05-07' }
```
Comment (floresv) [inline spec/factories/rents.rb]: and Faker? Same with the line before.

File: spec/rails_helper.rb [hunk 11]
```diff
@@ -2,7 +2,9 @@
 
 # This file is copied to spec/ when you run 'rails generate rspec:install'
 require 'spec_helper'
-require 'factories/factory_bot.rb'
+require 'factories/books.rb'
+require 'factories/users.rb'
+require 'factories/rents.rb'
```
Comment (floresv) [inline spec/rails_helper.rb]: mmm... I think this it's not scalable. What do you think?

**Review by floresv — CHANGES_REQUESTED** [review]
Remove development log file, and please check associations validations. What about rent without a book or user? Does that make sense?

File: app/models/rent.rb (hunk 8 above)
Comment (fcavazzoli) [reply app/models/rent.rb:5]: rubocop wants it

File: db/migrate/20190507152056_create_rents.rb (hunk 9 above)
Comment (fcavazzoli) [reply db/migrate/20190507152056_create_rents.rb]: cant be, im doing a validation when i create a new Rent

File: spec/rails_helper.rb (hunk 11 above)
Comment (fcavazzoli) [reply spec/rails_helper.rb]: not at all, sry my bad, solved

File: spec/factories/rents.rb (hunk 10 above)
Comment (fcavazzoli) [reply spec/factories/rents.rb]: solved

**Review by floresv — DISMISSED** [review]
LGTM

**Review by cquiro — APPROVED** [review]

**Review by cquiro — DISMISSED** [review]

**Review by floresv — DISMISSED** [review]
LGTM

**Review by floresv — APPROVED** [review]
LGTM

**Pull request merged**